eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -614,8 +614,8 @@ object Helpers {
 
       // remember we are looking at the remote commitment so IN for them is really OUT for us and vice versa
       val txes = remoteCommit.spec.htlcs.collect {
-        // incoming htlc for which we have the preimage: we spend it directly
-        case DirectedHtlc(OUT, add: UpdateAddHtlc) if preimages.exists(r => sha256(r) == add.paymentHash) => generateTx("claim-htlc-success")(Try {
+        // outgoing htlc for which we have the preimage: we spend it directly
```

Comment:
This is really an incoming HTLC (because it's outgoing in the *remote*'s commit). Yeah this is confusing :)
Maybe the comment could be updated to:
```scala
// incoming htlc for which we have the preimage: we spend it directly.
// NB: we are looking at the remote's commitment, from its point of view it's an outgoing htlc.
```